tests/drivers/candev/main.c

```diff
@@ -221,34 +300,42 @@ if (IS_ACTIVE(CONFIG_USE_LOOPBACK_MODE)) {
     candev->driver->set(candev, CANOPT_STATE, &mode, sizeof(mode));
 }
 
+/* Depending from the CAN controller used, this test example will provide different results.
+For MCP2515 standalone CAN controller, the last filter won't be applied as the first reception mailbox supports up to two filters
+For SAMD5x/E5x CAN controller, and with keeping the default parameters in candev_samd5x.h, the last filter won't be applied as the CAN controller supports up to 3 standard filters
+For SAMD5x/E5x CAN controller, if you increase the maximum capacity of the standard filters (as mentioned in Makefile.board.dep), the last filter can be applied correctly. */
+#if defined(MODULE_MCP2515)
     if (IS_ACTIVE(MCP2515_RECV_FILTER_EN)) {
+#endif
```

Comment:
```suggestion
    if (IS_ACTIVE(MCP2515_RECV_FILTER_EN) || !IS_ACTIVE(MODULE_MCP2515)) {
```

or like this?

Reply:
that would not do the same thing 

this does if is is the mcp.. then check if the filter thing is activated  else (not mcp.. ) just do the filter test 

the preprocessor removes that condition

Reply:
`MCP2515_RECV_FILTER_EN` is defined only for the mcp2515 so that it controls the CAN controller configuration a propos the CAN filters (Enable or disable). If no mcp2515 is used that condition is skipped as Karl mentioned